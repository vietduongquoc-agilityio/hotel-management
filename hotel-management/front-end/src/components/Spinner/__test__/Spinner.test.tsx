import { render, screen } from "@testing-library/react";
import LoadingSpinner from "..";

describe("LoadingSpinner", () => {
  it("should render a Box component with the correct styles", () => {
    const { container } = render(<LoadingSpinner />);
    const boxElement = screen.getByTestId("loading-spinner-box");

    expect(container).toMatchSnapshot();
    expect(boxElement).toBeInTheDocument();
    expect(boxElement).toHaveStyle({
      display: "flex",
      justifyContent: "center",
      alignItems: "center",
      height: "100px",
    });
  });
});
