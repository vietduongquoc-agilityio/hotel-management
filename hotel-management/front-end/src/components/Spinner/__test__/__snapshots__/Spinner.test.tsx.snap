// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner should render a Box component with the correct styles 1`] = `
<div>
  <div
    class="css-1lcwyqq"
    data-testid="loading-spinner-box"
  >
    <div
      class="chakra-spinner css-1y7joxr"
    >
      <span
        class="css-8b45rq"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
